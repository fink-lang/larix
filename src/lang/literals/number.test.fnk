{describe, it, expect, to_match_snapshot} = import '@fink/jest'

{parse, parse_expr} = import '../../'


describe:: 'numbers', fn:
  it:: 'parses int 12345', fn:
    expect::
      parse_expr:: ` 12345  `
      to_match_snapshot


  it:: 'parses hex, oct, bin', fn:
    expect::
      parse:: `
        0x123456789abcdef0
        0x123456789abcde
        0o12345670
        0b101010
      `
      to_match_snapshot


  it:: 'parses float 123.456', fn:
    expect::
      parse_expr:: `  123.456  `
      to_match_snapshot


  it:: 'parses float 123.456e10', fn:
    expect::
      parse:: `
        123.456e78
        123.456e+78
        123.456e-78
      `
      to_match_snapshot


  it:: 'errors when missing exponent', fn:
    expect::
      parse `123.456e * 78`, 'test.fnk'
      to_match_snapshot
