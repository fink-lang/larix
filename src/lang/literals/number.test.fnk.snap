// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`numbers errors when missing exponent 1`] = `
module (1:0-1:10)
  number (1:0-1:10) 123.456e*

error:
test.fnk:1:9
1| 123.456e * 78
            ^

Expected exponent.



`;

exports[`numbers parses float 123.456 1`] = `number (1:2-1:9) 123.456`;

exports[`numbers parses float 123.456e10 1`] = `
module (1:0-4:0)
  number (1:0-1:10) 123.456e78
  number (2:0-2:11) 123.456e+78
  number (3:0-3:11) 123.456e-78
`;

exports[`numbers parses hex, oct, bin 1`] = `
module (1:0-5:0)
  number (1:0-1:18) 0x123456789abcdef0
  number (2:0-2:16) 0x123456789abcde
  number (3:0-3:10) 0o12345670
  number (4:0-4:8) 0b101010
`;

exports[`numbers parses int 12345 1`] = `number (1:1-1:6) 12345`;
